includes/indices/class-algolia-posts-index.php

```diff
@@ -480,7 +481,7 @@ public function delete_item( $item ) {
 		}
 
 		if ( ! empty( $object_ids ) ) {
-			$this->get_index()->deleteObjects( $object_ids );
+			$wait ? $this->get_index()->deleteObjects( $object_ids )->wait() : $this->get_index()->deleteObjects( $object_ids );
 		}
```

Comment:
I think this will look better with a gate condition.

```
if ( empty( $object_ids ) ) {
	return;
}

if ( $wait ) {
	$this->get_index()->deleteObjects( $object_ids )->wait();
	return;
}

$this->get_index()->deleteObjects( $object_ids );
```